### DISCUSSION TITLE
Phasing out third-party cookies

### DISCUSSION THREAD

**Discussion by lvbn** [supabase/supabase#18900] [Questions] [answered]
Hello guys!

I am curious to know what are the implications of the third-party cookies phase out. As announced by Google, Chrome will no longer support third-party cookies:

[The phase out will be gradual and [starts from Q1 2024](https://privacysandbox.com/open-web/#the-privacy-sandbox-timeline) with disabling third-party cookies](https://developer.chrome.com/docs/privacy-sandbox/third-party-cookie-phase-out/?utm_source=devtools)

I am not sure if we will have to adapt our code, for example when using `createServerComponentClient({ cookies })`

I appreciate a lot your opinion and feedback! 

Thanks in advance!

**Comment by e-roy** [comment]
I'm very curious about this too.  I haven't seen anything yet that suggests that Supabase user's will not be effected by this.  It would be good to get some clarity on this, Chrome holds a large part of the [market share](https://gs.statcounter.com/browser-market-share).

**Comment by tomelliot** [comment]
I'd also appreciate a chime in from the supabase team. It'd be good to know what impact it will have on our projects.

**Comment by cody-lettau** [comment]
This is something that needs to be addressed here very soon. Google will be rolling this out in the near future and I do see warnings in Chrome regarding the use of 3rd party cookies when authenticating with Supabase using magic links and code auth (no custom domain setup as of yet).

@soedirgo @w3b6x9 @hf @kangmingtay -- It seems you guys are the most active contributors to supabase-js and auth-js, so tagging you to get your thoughts on this if possible!

**Comment by lvbn** [comment]
I am about to choose something else instead of Supabase for not knowing what is going to happen.

**Reply by cody-lettau** [reply]
Definitely will be looking into alternatives. This is something that could render every application using Auth worthless.

**Comment by kangmingtay** [comment, marked as answer]
Hey @lvbn @cody-lettau @tomelliot @e-roy and everyone else, kang ming from the auth team here - just to reassure everyone, Supabase Auth only uses 1st-party cookies for authentication and not 3rd party cookies.

In `supabase/ssr`, the cookies are being set on your server's domain, which is considered a 1st-party cookie.

[Definition](https://www.quantcast.com/wiki/term/cookie-first-party-vs-third-party/#:~:text=First%2Dparty%20cookies%20are%20available,be%20deleted%20by%20the%20user.):
> First-party cookies are available to the domain that created it, while third-party cookies are accessible on any website that loads the third-party’s server code. First-party and third-party cookies are supported by all browsers and can each be deleted by the user.

**Reply by cody-lettau** [reply]
Hi @kangmingtay - thanks for the reply! Is this the case when using the javascript client and allowing users to login (with password, otp, etc) -- basically using any of these flows defined [here](https://supabase.com/docs/reference/javascript/auth-signinwithpassword)?

I also want to confirm that this does not require use of a custom domain for our project - given the requests will be coming from a separate hosts (i.e. www.mydomain.com).

**Reply by GaryAustin1** [reply]
The REST API's don't use cookies to communicate with Supabase, everything is done with a JWT.  SSR uses cookies to store the session info so it can be used on your server with your browser clients.  Those cookies are not used by the Supabase servers.
